src/index/autotune.rs

```diff
@@ -0,0 +1,91 @@
+use std::ffi;
+use std::ptr;
+
+use super::*;
+
+use crate::error::Result;
+use crate::faiss_try;
+
+/// Uses a-priori knowledge on the Faiss indexes to extract tunable parameters.
+pub struct ParameterSpace {
+    inner: *mut FaissParameterSpace,
+}
+
+impl ParameterSpace {
+    /// Create a new instance
+    pub fn new() -> Result<Self> {
+        unsafe {
+            let mut inner = ptr::null_mut();
+            faiss_try(faiss_ParameterSpace_new(&mut inner))?;
+
+            Ok(ParameterSpace { inner })
+        }
+    }
+
+    /// Set one of the parameters
+    pub fn set_index_parameter(
+        &self,
+        index: &mut IndexImpl,
+        name: &ffi::CStr,
+        value: f64,
```

Comment:
This method can be even more flexible if the parameter value is anything which can be converted into an `f64`.

```rust
value: T,
...
)
where
    T: Into<f64>,
```

This way, one can pass values of many other types, including `i32`, `u32`, and `f32`.